json.go

```diff
@@ -12,21 +12,85 @@ import (
 // values as flag values. If the value is an array, the flag will be set
 // multiple times.
 func JSONParser(r io.Reader, set func(name, value string) error) error {
+	return NewJSONParser().Parse(r, set)
+}
+
+// JSONConfigFileParser is a parser for the JSON file format. Flags and their values
+// are read from the key/value pairs defined in the config file.
+// Nested objects and keys are concatenated with a delimiter to derive the
+// relevant flag name.
+type JSONConfigFileParser struct {
+	delimiter string
+}
+
+// NewJSONParser constructs and configures a JSONConfigFileParser using the provided options.
+func NewJSONParser(opts ...JSONOption) (c JSONConfigFileParser) {
+	c.delimiter = "."
+	for _, opt := range opts {
+		opt(&c)
+	}
+	return c
+}
```

Comment:
Should this style also be changed in the `fftoml` package?  
Is returning a pointer a benefit here?

Reply:
NewXxx should return a pointer to Xxx by convention.

Ideally yes it should be reflected in fftoml too, but as that would be a breaking change better to avoid it, unfortunately.